### PR TITLE
Preserve original exceptions in parameters

### PR DESCRIPTION
**Pull request by enumag**
Note: This is a quick edit using GitHub editor. I didn't really check or test it yet.

### CODE REVIEW

File: src/ActionEventEmitterEventStore.php [hunk 1]
```diff
@@ -221,11 +221,11 @@ public function appendTo(StreamName $streamName, Iterator $streamEvents): void
         $this->actionEventEmitter->dispatch($event);
 
         if ($event->getParam('streamNotFound', false)) {
-            throw StreamNotFound::with($streamName);
+            throw $event->getParam('streamNotFound');
         }
 
         if ($event->getParam('concurrencyException', false)) {
-            throw new ConcurrencyException();
+            throw $event->getParam('concurrencyException');
```
Comment (prolic) [inline src/ActionEventEmitterEventStore.php]: why get twice? just do `if ($exception = $event->getParam('concurrencyException', false)) {`

Comment (enumag) [reply src/ActionEventEmitterEventStore.php]: I don't really like assignments in ifs but sure, I'll change it.

**Comment by enumag** [comment]
@prolic Done. The coding style failure is not related.

**Comment by enumag** [comment]
There isn't much I can do in tests for this change - the exceptions didn't really change, except for their their callstack and I don't think we want to test that.

**Comment by prolic** [comment]
@enumag Can you also update `TransactionalActionEventEmitterEventStore` ?

**Comment by enumag** [comment]
Oh, I missed a class again lol. Sure, I'll update it tomorrow.

**Comment by prolic** [comment]
Thanks

**Comment by enumag** [comment]
@prolic Done.

**Pull request merged**